anaUltraScurve.py

```diff
@@ -0,0 +1,355 @@
+        vSum[i].GetYaxis().SetTitleOffset(1.5)
+        pass
+    if options.channels:
+        #vSum[i] = r.TH2D('vSum%i'%i,'vSum%i;Channels;VCal [DAC units]'%i,128,-0.5,127.5,256,-0.5,255.5)
+        vSum[i] = r.TH2D('vSum%i'%i,'VFAT %i;Channels;VCal [fC]'%i,128,-0.5,127.5,256,vToQm*-0.5+vToQb,vToQm*255.5+vToQb)        
+        vSum[i].GetYaxis().SetTitleOffset(1.5)
+        pass
+    if options.PanPin:
+        #vSum[i] = r.TH2D('vSum%i'%i,'vSum%i_0-63;63 - Panasonic Pin;VCal [DAC units]'%i,64,-0.5,63.5,256,-0.5,255.5)
+        vSum[i] = r.TH2D('vSum%i'%i,'VFAT %i_0-63;63 - Panasonic Pin;VCal [fC]'%i,64,-0.5,63.5,256,vToQm*-0.5+vToQb,vToQm*255.5+vToQb)
+        vSum[i].GetYaxis().SetTitleOffset(1.5)
+        #vSum2[i] = r.TH2D('vSum2_%i'%i,'vSum%i_64-127;127 - Panasonic Pin;VCal [DAC units]'%i,64,-0.5,63.5,256,-0.5,255.5)
+        vSum2[i] = r.TH2D('vSum2_%i'%i,'vSum%i_64-127;127 - Panasonic Pin;VCal [fC]'%i,64,-0.5,63.5,256,vToQm*-0.5+vToQb,vToQm*255.5+vToQb)
+        vSum2[i].GetYaxis().SetTitleOffset(1.5)
+        pass
+    for ch in range (0,128):
+        vScurves[i].append(r.TH1D('Scurve_%i_%i'%(i,ch),'Scurve_%i_%i;VCal [DAC units]'%(i,ch),256,-0.5,255.5))
+        vthr_list[i].append(0)
+        trim_list[i].append(0)
+        trimrange_list[i].append(0)
+        pass
+    pass
+
+if options.SaveFile:
+    scanFits = fitScanData(filename+'.root')
+    pass
+
+for event in inF.scurveTree:
+    #strip = chToStrLUT[event.vfatN][event.vfatCH]
+    strip = lookup_table[event.vfatN][event.vfatCH]
+    pan_pin = pan_lookup[event.vfatN][event.vfatCH]
+    if not (options.channels or options.PanPin):
+        #vSum[event.vfatN].Fill(strip,event.vcal,event.Nhits)
+        vSum[event.vfatN].Fill(strip,vToQm*event.vcal+vToQb,event.Nhits)
+        pass
+    if options.channels:
+        #vSum[event.vfatN].Fill(event.vfatCH,event.vcal,event.Nhits)
+        vSum[event.vfatN].Fill(event.vfatCH,vToQm*event.vcal+vToQb,event.Nhits)
+        pass
+    if options.PanPin:
+        if (pan_pin < 64):
+            #vSum[event.vfatN].Fill(63-pan_pin,event.vcal,event.Nhits)
+            vSum[event.vfatN].Fill(63-pan_pin,vToQm*event.vcal+vToQb,event.Nhits)
+            pass
+        else:
+            #vSum2[event.vfatN].Fill(127-pan_pin,event.vcal,event.Nhits)
+            vSum2[event.vfatN].Fill(127-pan_pin,vToQm*event.vcal+vToQb,event.Nhits)
+            pass
+        pass
+    x = vScurves[event.vfatN][event.vfatCH].FindBin(event.vcal)
+    vScurves[event.vfatN][event.vfatCH].SetBinContent(x, event.Nhits)
+    r.gStyle.SetOptStat(1111111)
+    vthr_list[event.vfatN][event.vfatCH] = event.vthr
+    trim_list[event.vfatN][event.vfatCH] = event.trimDAC
+    trimrange_list[event.vfatN][event.vfatCH] = event.trimRange
+    pass
+
+if options.SaveFile:
+    masks = []
+    fitSums = {}
+    for vfat in range (0,24):
+        fitThr = []        
+        fitENC = []
+        strList = []
+        masks.append([])
+        for ch in range (0, 128):
```

Comment:
that is right, strList fixes it, I forgot I did that. We should add panList and chList so we can control which we pass to build the TGraphErrors and can control the units of the x-axis then. As the code is now, it will always be in strip units.